### PR TITLE
Add Clue Scroll/Barrows to screenshot plugin

### PR DESCRIPTION
**Pull request by Sethtroll**

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java [hunk 1]
```diff
@@ -226,6 +279,25 @@ public void hideWidgets(WidgetHiddenChanged event)
...
+			{
+				if (clueType != null && clueNumber != null)
+				{
+					fileName = Character.toUpperCase(clueType.charAt(0)) + clueType.substring(1) + "(" + clueNumber + ")";
+					clueType = null;
+					clueNumber = null;
+					break;
+				}
+			}
+			case BARROWS_REWARD_GROUP_ID:
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java]: this should not fallthrough

File: runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java [hunk 2]
```diff
@@ -226,6 +279,25 @@ public void hideWidgets(WidgetHiddenChanged event)
 				fileName = text.substring(19, text.length() - 1);
 				break;
 			}
+			case CLUE_SCROLL_REWARD_GROUP_ID:
+			{
+				if (clueType != null && clueNumber != null)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java]: false case should return

File: runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java [hunk 3]
```diff
@@ -226,6 +279,25 @@ public void hideWidgets(WidgetHiddenChanged event)
...
+				{
+					fileName = Character.toUpperCase(clueType.charAt(0)) + clueType.substring(1) + "(" + clueNumber + ")";
+					clueType = null;
+					clueNumber = null;
+					break;
+				}
+			}
+			case BARROWS_REWARD_GROUP_ID:
+			{
+				if (barrowsNumber != null)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java]: false case should return

File: runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java [hunk 4]
```diff
@@ -179,6 +194,44 @@ private void removeButtonFromTitlebar()
 		});
 	}
 
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java]: Logic like this is very easy to add tests for

File: runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java [hunk 5]
```diff
@@ -92,6 +100,13 @@
 	private static final DateFormat DATE_FORMAT = new SimpleDateFormat("MMM. dd, yyyy", Locale.US);
 	private static final DateFormat TIME_FORMAT = new SimpleDateFormat("h.m.s a d MMM. yyyy", Locale.US);
 
+	private static final Pattern NUMBER_PATTERN = Pattern.compile("([0-9]+)");
+
+	private static String clueType;
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java]: These should be nonstatic

File: runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java [hunk 6]
```diff
@@ -92,6 +100,13 @@
 	private static final DateFormat DATE_FORMAT = new SimpleDateFormat("MMM. dd, yyyy", Locale.US);
 	private static final DateFormat TIME_FORMAT = new SimpleDateFormat("h.m.s a d MMM. yyyy", Locale.US);
 
+	private static final Pattern NUMBER_PATTERN = Pattern.compile("([0-9]+)");
+
+	public String clueType;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java]: These instance fields should be private

**Pull request merged**